jplag/src/main/java/de/jplag/reportingV2/reportobject/ReportObjectFactory.java

```diff
@@ -0,0 +1,209 @@
+        }
+        HashSet<String> excludedFileNames = new HashSet<>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(options.getExclusionFileName(), JPlagOptions.CHARSET));
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                excludedFileNames.add(line.trim());
+            }
+
+            reader.close();
+        } catch (IOException exception) {
+            System.out.println("Could not read exclusion file: " + exception.getMessage());
+        }
+
+        return new ArrayList<>(excludedFileNames);
+    }
+
+    /**
+     * Converts files of a submission to FilesOFSubmission DTO.
+     * @return A list containing FilesOfSubmission DTOs.
+     */
+    private static List<FilesOfSubmission> getFilesForSubmission(Submission submission) {
+        return submission.getFiles().stream().map(f -> new FilesOfSubmission(f.getName(), readFileLines(f))).collect(Collectors.toList());
+    }
+
+    /**
+     * Converts a JPlag Match object to a Match DTO.
+     * @param comparison The comparison from which the match originates.
+     * @param match The match to be converted.
+     * @param usesIndex Indicates whether the language uses indexes.
+     * @return A Match DTO.
+     */
+    private static Match convertMatchToReportMatch(JPlagComparison comparison, de.jplag.Match match, Boolean usesIndex) {
+        TokenList tokensFirst = comparison.getFirstSubmission().getTokenList();
+        TokenList tokensSecond = comparison.getSecondSubmission().getTokenList();
+        Token startFirstToken = tokensFirst.getToken(match.getStartOfFirst());
+        Token endFirstToken = tokensFirst.getToken(match.getStartOfFirst() + match.getLength() - 1);
+        Token startSecondToken = tokensSecond.getToken(match.getStartOfSecond());
+        Token endSecondToken = tokensSecond.getToken(match.getStartOfSecond() + match.getLength() - 1);
+
+        int startFirst = usesIndex ? startFirstToken.getIndex() : startFirstToken.getLine();
+        int endFirst = usesIndex ? endFirstToken.getIndex() : endFirstToken.getLine();
+        int startSecond = usesIndex ? startSecondToken.getIndex() : startSecondToken.getLine();
+        int endSecond = usesIndex ? endSecondToken.getIndex() : endSecondToken.getLine();
+        int tokens = match.getLength();
+
+        return new Match(startFirstToken.file, startSecondToken.file, startFirst, endFirst, startSecond, endSecond, tokens);
+    }
+
+    // TODO implement after PR Readd clustering #281
+    private static List<Cluster> getClusters(JPlagResult result) {
+        // List<ClusteringResult<Submission>> clusters = result.getClusteringResult();
+        // return clusters.map( c -> new Cluster(getAvgSimilarity, getStrength, c.getMembers().map(Submission::getName)))
+        return List.of();
+    }
+
+    private static List<String> readFileLines(File file) {
+        ArrayList<String> lines = new ArrayList<>();
```

Comment:
Replace static type with List

Reply:
This is actually a copied method from another part of JPlag due to missing attributes. See #248 . It won't be present in the final version.